cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/InMemoryDatasetFramework.java

```diff
@@ -257,6 +258,12 @@ public void addInstance(String datasetType, DatasetId datasetInstanceId,
   }
 
   @Override
+  public void addInstance(String datasetTypeName, DatasetId datasetInstanceId, DatasetProperties props,
+                          @Nullable KerberosPrincipalId ownerPrincipal) throws DatasetManagementException, IOException {
+    throw new UnsupportedOperationException("Creating dataset with owner is not supported");
```

Comment:
In InMemory/Standalone (unless where required to run unit tests) we don't support UGI/Kerberos that is why we don't want to support creating things with an owner.